### ISSUE TITLE
PublicKey details (including FingerPrint) change from 5.5.0 to >=5.6.0

### ISSUE THREAD

**Issue by FuncLun** [mattosaurus/PgpCore#168]
The PublicKey information is different when comparing 5.5.0 and 5.6.0 and above. Specifically, the Fingerprint is not the same after upgrading.

Example: use stripe's "PGP Management Key: from here: https://stripe.com/docs/account/data-migrations/pan-import#migration-pgp-key

They publish their finger print as: `AEBF 7C48 38C4 4D2F DC99 A3F9 9C78 B762 0C1E 99AD`

Using the following:
```c#
var pgpKeyString = "-----BEGIN PGP PUBLIC KEY BLOCK-----"; //The full key omitted for readability
var pgpKey = new EncryptionKeys(pgpKeyString);
var fingerprint = BitConverter.ToString(pgpKey.PublicKey.GetFingerprint());
```

Values from 5.5.0, which are expected:
```
pgpKey.CreationTime == {7/3/2019 10:32:01 PM}
pgpKey.KeyId == -7171780774849504851
pgpKey.ValidDays == 1449
fingerprint == "AE-BF-7C-48-38-C4-4D-2F-DC-99-A3-F9-9C-78-B7-62-0C-1E-99-AD"
```
Values from 5.6.0, which are unexpected:
```
pgpKey.CreationTime == {3/29/2022 5:14:51 PM}
pgpKey.KeyId == 3868062543259959644
pgpKey.ValidDays == 450
fingerprint == "B3-54-01-FA-B0-5F-51-DB-E5-AB-A1-4C-35-AE-1E-63-A4-01-01-5C"
```

`pgpKey.PublicKeys.Count()` is 1 for both versions - so there shouldn't confusion with mixing up different keys.

One change between 5.5.0 and 5.6.0 was an upgrade to Portable.BouncyCastle 1.8.9 to 1.9.0. I've used PgpCore 5.5.0 with Portable.BouncyCastle 1.9.0 and it gave me the expected results.

I could not test 5.5.1 as it is no longer published.

5.7.0 gives the same results at 5.6.0.

**Comment by mattosaurus** [comment]
It looks like this is due to the changes that were made in order to help pick the most appropriate key for encryption/signing.

As part of this we score each key for the task and take the highest scoring.

https://github.com/mattosaurus/PgpCore/blob/9799127c779dd319522922e0e3a4f37d5e0f639f/PgpCore/Utilities.cs#L869-L886

The Stripe key actually contains 3 keys, a master key with the encryption flag set (this is your expected key), an encryption key with both the EncryptComms and EncryptStorage flags set (this is the returned key) and a plain encryption key.

The issue is the weighting we attach to these keys to determine which to pick. I'll try and weight this more towards the master as I agree that's the correct one to be using.

**Comment by mattosaurus** [comment]
Not sure if this is the best solution but I've added an additional property `MasterKey` to the `EncryptionKeys` object, this is the key that you're expecting.

The alternative is to change the way the scoring works for picking the encryption key but I feel that may have other downstream consequences. Encryption will still occur using the encryption subkey (not the master key) but will be decryptable by the private key.

I've created a draft PR request here

https://github.com/mattosaurus/PgpCore/pull/172

Let me know if you think this solves the issue or if you have any concerns.

**Comment by FuncLun** [comment]
Yes, exposing the MasterKey would work.

I was able to use your code to get what I need using v5.7.0, but, providing the master key as a property would be better.

```c#
var masterKey = Utilities.ReadAllKeyRings(pgpKeyString.GetStream())
        .First()
        .GetPublicKeys()
        .Cast<PgpPublicKey>()
        .Single(k => k.IsMasterKey);

var fingerprint = BitConverter.ToString(masterKey.GetFingerprint());
```

**Comment by mattosaurus** [comment]
This is hopefully fixed in v5.9.0 with #172
